### PR TITLE
Fix the permissions

### PR DESCRIPTION
**Pull request by RoySegall**
#140

### CODE REVIEW

File: biblio.module [hunk 1]
```diff
@@ -463,25 +436,77 @@ function biblio_type_delete($type) {
...
+/**
+ * Access function for the biblio contributor entity.
+ *
+ * @param $op
+ *  The type of the operation.
+ * @param Biblio|BiblioContributor $entity
+ *  The entity object.
+ * @param $account
+ *  The user object.
+ * @param $entity_type
```
Comment (amitaibu) [inline biblio.module:503]: is this still needed?

File: biblio.module [hunk 2]
```diff
@@ -463,25 +436,77 @@ function biblio_type_delete($type) {
...
+    'update' => 'edit biblio content',
+    'delete' => 'delete biblio content',
+    'view' => 'access biblio content',
+  );
+
+  $string = $strings[$op];
+
+  // When checking the access on the biblio entity we need to check if the user
+  // can delete/update entries which created by himself/other users.
+  if (is_object($entity) && $entity_type == 'biblio') {
```
Comment (amitaibu) [inline biblio.module]: `$entity_type == 'biblio'` can probably be removed.

File: biblio.module [hunk 3]
```diff
@@ -463,25 +436,77 @@ function biblio_type_delete($type) {
...
+ * @param $account
+ *  The user object.
+ * @param $entity_type
+ *  The entity type.
+ *
+ * @return bool
+ *  Return TRUE/FALSE if the user has the privilege for this action.
  */
-function biblio_entity_access($op, $type = NULL, $account = NULL) {
+function biblio_access($op, $entity = NULL, $account = NULL, $entity_type = NULL) {
```
Comment (amitaibu) [inline biblio.module]: You can type hint the $entity (`Biblio $biblio`)

**Comment by amitaibu** [comment]
@RoySegall needs re-roll.

**Comment by RoySegall** [comment]
@amitaibu Re rolled

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**